test/e2e/external_auth_complete_cluster.go

```diff
@@ -0,0 +1,242 @@
+				groupsClaim               = "groups"
+				externalAuthComponentName = "console"
+				externalAuthComponentNS   = "openshift-console"
+			)
+
+			By("creating resource group")
+			resourceGroup, err := tc.NewResourceGroup(ctx, "external-auth", region)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("provisioning infra")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "infra",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/customer-infra.json")),
+				map[string]interface{}{
+					"customerNsgName":        customerNSGName,
+					"customerVnetName":       customerVnetName,
+					"customerVnetSubnetName": customerVnetSubnetName,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating HCP cluster (injecting MAJOR.MINOR version)")
+			rawClusterTpl := framework.Must(
+				TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/cluster.json"),
+			)
+			ver := strings.TrimSpace(os.Getenv("HCP_OPENSHIFT_VERSION"))
+			if ver == "" {
+				ver = "4.19" // default MAJOR.MINOR required by API
+			}
+			clusterTpl := applyVersionToTemplate(rawClusterTpl, ver)
+
+			managedRG := framework.SuffixName(*resourceGroup.Name, "-managed", 64)
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "hcp-cluster",
+				clusterTpl,
+				map[string]interface{}{
+					"nsgName":                  customerNSGName,
+					"vnetName":                 customerVnetName,
+					"subnetName":               customerVnetSubnetName,
+					"clusterName":              customerClusterName,
+					"managedResourceGroupName": managedRG,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating node pool")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "node-pool",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/nodepool.json")),
+				map[string]interface{}{
+					"clusterName":  customerClusterName,
+					"nodePoolName": customerNodePoolName,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("applying ExternalAuth config via RP (using Entra app from Step 1)")
+
+			// 1) Load Entra creds JSON (local loader)
+			secretPath := strings.TrimSpace(os.Getenv("ENTRA_E2E_SECRET_PATH"))
+			if secretPath == "" {
+				secretPath = "test/e2e/out/entra_app_secret.json"
+			}
+			sf := mustLoadEntraSecretStep2(secretPath)
+
+			// 2) Build issuer/audience/client from Entra
+			issuerURL := fmt.Sprintf("https://login.microsoftonline.com/%s/v2.0", sf.TenantID)
+			externalAuthClientID := sf.ClientID
+			cliAudience := sf.ClientID
+			externalAuthClientSecret := ""
```

Comment:
clients are defined out of band?  we'll want to create these unless there's a security boundary reason we cannot.